browser/src/client_win.cc

```diff
@@ -148,6 +148,65 @@ NTSTATUS WaitForPipeAvailability(const UNICODE_STRING& path) {
   return sts;
 }
 
+// Reads the next message from the pipe and returns a buffer of chars.
+// This function is synchronous.
+std::vector<char> ReadNextMessageFromPipe(
+    HANDLE pipe,
+    OVERLAPPED* overlapped) {
+  DWORD err = ERROR_SUCCESS;
+  std::vector<char> buffer(kBufferSize);
+  char* p = buffer.data();
+  int final_size = 0;
+  while (true) {
+    DWORD read;
+    if (ReadFile(pipe, p, kBufferSize, &read, overlapped)) {
+      final_size += read;
+      break;
+    } else {
+      err = GetLastError();
+      if (err == ERROR_IO_PENDING) {
+        if (GetOverlappedResult(pipe, overlapped, &read, /*wait=*/TRUE)) {
+          final_size += read;
+          break;
+        } else {
```

Comment:
Can you add a comment to explain this part?

Reply:
Done